doc/api/axis_api.rst

```diff
@@ -1,12 +1,954 @@
-****
-axis
-****
+===================
+ axis and tick API
+===================
 
-
-:mod:`matplotlib.axis`
-======================
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+   :backlinks: entry
 
 .. automodule:: matplotlib.axis
-   :members:
-   :undoc-members:
-   :show-inheritance:
+
+Inheritance
+===========
+
+.. inheritance-diagram:: Tick Ticker XAxis YAxis XTick YTick
+   :private-bases:
```

Comment:
Because I thought it went better at the top in `Artist` and better at the bottom here.  The reason for that is that I would expect users to frequently directly use the `Axis` classes, but not the `Tick` classes, hence I put the diagram at the bottom.  Having now written out this justification, that makes me think that `Axis` and `Tick` should get their own pages.